plugins/task-plugin/src/che-workspace-client.ts

```diff
@@ -32,6 +33,10 @@ export class CheWorkspaceClient {
     }
 
     async getCommands(): Promise<cheApi.workspace.Command[]> {
+        if (this.cachedCommands.length > 0) {
+            return this.cachedCommands;
+        }
+
```

Comment:
I would avoid to cache it there.
Probably better to cache getCurrentWorkspace if config is not changed

Reply:
Agree, what about caching on che task provider level?

Reply:
cache is good but we need to invalidate cache when workspace configuration is changed

Reply:
We don't have events about changing workspace config, but workspace and plugins will be restarted if configuration is changed. Is it correct?